sympy/solvers/solveset.py

```diff
@@ -1325,6 +1329,65 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_lambert(f, symbol):
+    return any(isinstance(arg, (Pow, exp, log)) for arg in _term_factors(f))
```

Comment:
Please document all the methods so that the reviewer understands the intention.

Reply:
Sure I will do that. I have added it as a todo.